pages/querying/read-and-modify-data.md

```diff
@@ -532,3 +532,149 @@ relationships:
 ```cypher
 MATCH (n:WrongLabel) REMOVE n:WrongLabel, n.property;
 ```
+
+## Periodic execution
+
+Write queries use additional memory overhead when modifying data, since the query needs 
+to be reverted to its original state if it fails. The memory overhead is linear to the number
+of updates performed during the queries. These updates during query execution are called Delta
+objects, which you can refer [to on this page](/fundamentals/storage-memory-usage#in-memory-transactional-storage-mode-default). The number of created Delta
+objects grows during the execution of write queries, and that takes additional memory space, which
+can be significant if the user is restricted with RAM memory. Although Delta objects are temporary, and
+are being deleted by the GC at the transaction end, that is sometimes not enough as there is already a
+huge amount of deltas present in the system during the end.
+In the following example:
+
+```cypher
+MATCH (n) DETACH DELETE n;
+```
+
+The amount of created Delta overhead is the total number of nodes and edges in the graph. Every
+delta is around 56 bytes in size. For a dataset of 40 million entities, it piles up to 2.25GB, which

```

Comment:
Don't say "piles" - that's too informal. Use a more precise phrase such as "adds up to" or "accumulates".